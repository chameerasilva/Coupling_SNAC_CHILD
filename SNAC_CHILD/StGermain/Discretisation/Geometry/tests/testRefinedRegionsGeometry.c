/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
**
** Copyright (C), 2003, Victorian Partnership for Advanced Computing (VPAC) Ltd, 110 Victoria Street, Melbourne, 3053, Australia.
**
** Authors:
**	Stevan M. Quenette, Senior Software Engineer, VPAC. (steve@vpac.org)
**	Patrick D. Sunter, Software Engineer, VPAC. (pds@vpac.org)
**	Luke J. Hodkinson, Computational Engineer, VPAC. (lhodkins@vpac.org)
**	Siew-Ching Tan, Software Engineer, VPAC. (siew@vpac.org)
**	Alan H. Lo, Computational Engineer, VPAC. (alan@vpac.org)
**	Raquibul Hassan, Computational Engineer, VPAC. (raq@vpac.org)
**
**  This library is free software; you can redistribute it and/or
**  modify it under the terms of the GNU Lesser General Public
**  License as published by the Free Software Foundation; either
**  version 2.1 of the License, or (at your option) any later version.
**
**  This library is distributed in the hope that it will be useful,
**  but WITHOUT ANY WARRANTY; without even the implied warranty of
**  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
**  Lesser General Public License for more details.
**
**  You should have received a copy of the GNU Lesser General Public
**  License along with this library; if not, write to the Free Software
**  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
**
** $Id: testRefinedRegionsGeometry.c 3462 2006-02-19 06:53:24Z WalterLandry $
**
**~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/

#include <mpi.h>
#include "Base/Base.h"

#include "Discretisation/Geometry/Geometry.h"

#include <stdio.h>


int main( int argc, char* argv[] ) {
	Dictionary*	dictionary;
	RefinedRegionsGeometry*	geometry;
	Coord_List	points;
	Index		i;
	MPI_Comm	CommWorld;
	int		rank;
	int		numProcessors;
	Dictionary_Entry_Value*	refinedRegion;
	Dictionary_Entry_Value*	refinedRegions;
	
	/* Initialise MPI, get world info */
	MPI_Init( &argc, &argv );
	MPI_Comm_dup( MPI_COMM_WORLD, &CommWorld );
	MPI_Comm_size( CommWorld, &numProcessors );
	MPI_Comm_rank( CommWorld, &rank );
	
	Base_Init( &argc, &argv );
	
	DiscretisationGeometry_Init( &argc, &argv );
	MPI_Barrier( CommWorld ); /* Ensures copyright info always come first in output */
	
	dictionary = Dictionary_New();
	Dictionary_Add( dictionary, "meshSizeI", Dictionary_Entry_Value_FromUnsignedInt( 11 ) );
	Dictionary_Add( dictionary, "meshSizeJ", Dictionary_Entry_Value_FromUnsignedInt( 5 ) );
	Dictionary_Add( dictionary, "meshSizeK", Dictionary_Entry_Value_FromUnsignedInt( 1 ) );
	refinedRegions = Dictionary_Entry_Value_NewList();
	Dictionary_Add( dictionary, "RefinedRegions", refinedRegions ); 
	refinedRegion = Dictionary_Entry_Value_NewStruct();
	Dictionary_Entry_Value_AddMember( refinedRegion, "dim", Dictionary_Entry_Value_FromUnsignedInt( 0 ) );
	Dictionary_Entry_Value_AddMember( refinedRegion, "regionStart", Dictionary_Entry_Value_FromDouble( 0.0 ) );
	Dictionary_Entry_Value_AddMember( refinedRegion, "regionEnd", Dictionary_Entry_Value_FromDouble( 0.1 ) );
	Dictionary_Entry_Value_AddMember( refinedRegion, "refinementFactor", Dictionary_Entry_Value_FromUnsignedInt( 2 ) );
	Dictionary_Entry_Value_AddElement( refinedRegions, refinedRegion );
	refinedRegion = Dictionary_Entry_Value_NewStruct();
	Dictionary_Entry_Value_AddMember( refinedRegion, "dim", Dictionary_Entry_Value_FromUnsignedInt( 1 ) );
	Dictionary_Entry_Value_AddMember( refinedRegion, "regionStart", Dictionary_Entry_Value_FromDouble( 0.75 ) );
	Dictionary_Entry_Value_AddMember( refinedRegion, "regionEnd", Dictionary_Entry_Value_FromDouble( 1.0 ) );
	Dictionary_Entry_Value_AddMember( refinedRegion, "refinementFactor", Dictionary_Entry_Value_FromUnsignedInt( 3 ) );
	Dictionary_Entry_Value_AddElement( refinedRegions, refinedRegion );

	
	geometry = RefinedRegionsGeometry_New( "refinedRegionsGeometry", dictionary );

	printf( "Total point test:\n" );
	points = Memory_Alloc_Array( Coord, geometry->pointCount, "points" );
	geometry->buildPoints( geometry, points );
	for( i = 0; i < geometry->pointCount; i++ ) {
		printf( "\tPoint %u : ", i );
		printf( "{ %g, %g, %g }\n", points[i][0], points[i][1], points[i][2] );
	}
	printf( "\n" );

	printf( "Individual point test:\n" );
	for( i = 0; i < geometry->pointCount; i++ ) {
		Coord point;
		
		printf( "\tPoint %u : ", i );
		geometry->pointAt( geometry, i, point );
		printf( "{ %g, %g, %g }\n", point[0], point[1], point[2] );
	}
	printf( "\n" );
	
	if( points )
		Memory_Free( points );
	
	Stg_Class_Delete( dictionary );
	
	DiscretisationGeometry_Finalise();
	
	Base_Finalise();
	
	/* Close off MPI */
	MPI_Finalize();
	
	return 0;
}
