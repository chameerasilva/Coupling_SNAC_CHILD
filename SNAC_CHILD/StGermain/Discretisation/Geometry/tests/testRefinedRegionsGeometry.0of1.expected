StGermain Framework. Copyright (C) 2003-2005 VPAC.
Total point test:
	Point 0 : { 0, 0, 0 }
	Point 1 : { 0.05, 0, 0 }
	Point 2 : { 0.1, 0, 0 }
	Point 3 : { 0.2125, 0, 0 }
	Point 4 : { 0.325, 0, 0 }
	Point 5 : { 0.4375, 0, 0 }
	Point 6 : { 0.55, 0, 0 }
	Point 7 : { 0.6625, 0, 0 }
	Point 8 : { 0.775, 0, 0 }
	Point 9 : { 0.8875, 0, 0 }
	Point 10 : { 1, 0, 0 }
	Point 11 : { 0, 0.375, 0 }
	Point 12 : { 0.05, 0.375, 0 }
	Point 13 : { 0.1, 0.375, 0 }
	Point 14 : { 0.2125, 0.375, 0 }
	Point 15 : { 0.325, 0.375, 0 }
	Point 16 : { 0.4375, 0.375, 0 }
	Point 17 : { 0.55, 0.375, 0 }
	Point 18 : { 0.6625, 0.375, 0 }
	Point 19 : { 0.775, 0.375, 0 }
	Point 20 : { 0.8875, 0.375, 0 }
	Point 21 : { 1, 0.375, 0 }
	Point 22 : { 0, 0.75, 0 }
	Point 23 : { 0.05, 0.75, 0 }
	Point 24 : { 0.1, 0.75, 0 }
	Point 25 : { 0.2125, 0.75, 0 }
	Point 26 : { 0.325, 0.75, 0 }
	Point 27 : { 0.4375, 0.75, 0 }
	Point 28 : { 0.55, 0.75, 0 }
	Point 29 : { 0.6625, 0.75, 0 }
	Point 30 : { 0.775, 0.75, 0 }
	Point 31 : { 0.8875, 0.75, 0 }
	Point 32 : { 1, 0.75, 0 }
	Point 33 : { 0, 0.875, 0 }
	Point 34 : { 0.05, 0.875, 0 }
	Point 35 : { 0.1, 0.875, 0 }
	Point 36 : { 0.2125, 0.875, 0 }
	Point 37 : { 0.325, 0.875, 0 }
	Point 38 : { 0.4375, 0.875, 0 }
	Point 39 : { 0.55, 0.875, 0 }
	Point 40 : { 0.6625, 0.875, 0 }
	Point 41 : { 0.775, 0.875, 0 }
	Point 42 : { 0.8875, 0.875, 0 }
	Point 43 : { 1, 0.875, 0 }
	Point 44 : { 0, 1, 0 }
	Point 45 : { 0.05, 1, 0 }
	Point 46 : { 0.1, 1, 0 }
	Point 47 : { 0.2125, 1, 0 }
	Point 48 : { 0.325, 1, 0 }
	Point 49 : { 0.4375, 1, 0 }
	Point 50 : { 0.55, 1, 0 }
	Point 51 : { 0.6625, 1, 0 }
	Point 52 : { 0.775, 1, 0 }
	Point 53 : { 0.8875, 1, 0 }
	Point 54 : { 1, 1, 0 }

Individual point test:
	Point 0 : { 0, 0, 0 }
	Point 1 : { 0.05, 0, 0 }
	Point 2 : { 0.1, 0, 0 }
	Point 3 : { 0.2125, 0, 0 }
	Point 4 : { 0.325, 0, 0 }
	Point 5 : { 0.4375, 0, 0 }
	Point 6 : { 0.55, 0, 0 }
	Point 7 : { 0.6625, 0, 0 }
	Point 8 : { 0.775, 0, 0 }
	Point 9 : { 0.8875, 0, 0 }
	Point 10 : { 1, 0, 0 }
	Point 11 : { 0, 0.375, 0 }
	Point 12 : { 0.05, 0.375, 0 }
	Point 13 : { 0.1, 0.375, 0 }
	Point 14 : { 0.2125, 0.375, 0 }
	Point 15 : { 0.325, 0.375, 0 }
	Point 16 : { 0.4375, 0.375, 0 }
	Point 17 : { 0.55, 0.375, 0 }
	Point 18 : { 0.6625, 0.375, 0 }
	Point 19 : { 0.775, 0.375, 0 }
	Point 20 : { 0.8875, 0.375, 0 }
	Point 21 : { 1, 0.375, 0 }
	Point 22 : { 0, 0.75, 0 }
	Point 23 : { 0.05, 0.75, 0 }
	Point 24 : { 0.1, 0.75, 0 }
	Point 25 : { 0.2125, 0.75, 0 }
	Point 26 : { 0.325, 0.75, 0 }
	Point 27 : { 0.4375, 0.75, 0 }
	Point 28 : { 0.55, 0.75, 0 }
	Point 29 : { 0.6625, 0.75, 0 }
	Point 30 : { 0.775, 0.75, 0 }
	Point 31 : { 0.8875, 0.75, 0 }
	Point 32 : { 1, 0.75, 0 }
	Point 33 : { 0, 0.875, 0 }
	Point 34 : { 0.05, 0.875, 0 }
	Point 35 : { 0.1, 0.875, 0 }
	Point 36 : { 0.2125, 0.875, 0 }
	Point 37 : { 0.325, 0.875, 0 }
	Point 38 : { 0.4375, 0.875, 0 }
	Point 39 : { 0.55, 0.875, 0 }
	Point 40 : { 0.6625, 0.875, 0 }
	Point 41 : { 0.775, 0.875, 0 }
	Point 42 : { 0.8875, 0.875, 0 }
	Point 43 : { 1, 0.875, 0 }
	Point 44 : { 0, 1, 0 }
	Point 45 : { 0.05, 1, 0 }
	Point 46 : { 0.1, 1, 0 }
	Point 47 : { 0.2125, 1, 0 }
	Point 48 : { 0.325, 1, 0 }
	Point 49 : { 0.4375, 1, 0 }
	Point 50 : { 0.55, 1, 0 }
	Point 51 : { 0.6625, 1, 0 }
	Point 52 : { 0.775, 1, 0 }
	Point 53 : { 0.8875, 1, 0 }
	Point 54 : { 1, 1, 0 }

